src/torrent.cpp

```diff
@@ -2999,7 +3005,11 @@ bool is_downloading_state(int const st)
 				if (!aep.can_announce(now, is_seed(), ae.fail_limit))
 				{
 					// this counts
-					if (aep.is_working()) state.sent_announce = true;
+					if (aep.is_working())
+					{
+						state.sent_announce = true;
+						state.done = true;
```

Comment:
I think this needs only set `done` when neither `announce_to_all_tiers` nor `announce_to_all_trackers` are set (i.e. same as the existing logic for setting `done`). I also think this is the only change need to fix #4302. The other `state.done = true` instances added by this PR should not be there. We should not abandon announcing on an endpoint just because one tracker isn't reachable via that endpoint.

Reply:
I’ll test that. Regarding stopping without trying to announce all endpoints; right now content busing to the new tracker ends up announcing the first endpoint again. I didn’t feel confident I fully understood this logic to fix that. But maybe I should just make sure *all* cases are covered by the test

Reply:
I implemented your suggestion. it turns out the call to `prioritize_tracker()` had to be taken out as well, to support the case where different endpoints end up using different trackers. I think the call to `deprioritize_tracker()` should be enough though